src/js/components/Button/Button.js

```diff
@@ -67,9 +67,13 @@ class Button extends Component {
 
     let buttonIcon = icon;
     // only change color if user did not specify the color themselves...
-    if (primary && icon && !icon.props.color) {
+    if (icon && !icon.props.color) {
+      const themeColor = primary
+        ? theme.button.primary.color
```

Comment:
This doesn't look good to me .
![screenshot from 2019-02-27 23-51-51](https://user-images.githubusercontent.com/3874836/53528826-08bf6f80-3aeb-11e9-83f1-5707e4f786a1.png)

Haven't looked where the color is handled for the primary but it seems that something is missing here